src/include/duckdb/catalog/catalog_set.hpp

```diff
@@ -119,4 +123,29 @@ class CatalogSet {
 	unique_ptr<DefaultGenerator> defaults;
 };
 
+//! Class responsible to remove entries from the catalog.
+//! When deleting many types of errors can be thrown, since we want to avoid try/catch blocks
+//! this class makes sure that whatever elements were modified are returned to a correct state
+//! when exceptions are thrown.
+//! The idea here is to use RAII (Resource acquisition is initialization) to mimic a try/catch/finally block.
+//! If any exception is raised during DropEntryDependencies, then the destructor of EntryDropper will be called
+//! and the entry will return to its previous state.
+class EntryDropper {
```

Comment:
This class can be moved to the C++ file, as it isn't used outside of `catalog_set.cpp`, right?